### PR TITLE
failing test and updated requirements for running tests

### PR DESCRIPTION
**Pull request by caddac**
Added a test that shows my error case, also updated the requirements.txt with the minimums I needed to run the tests using a venv.

### CODE REVIEW

File: pandas_schema/validation.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def get_errors(self, series: pd.Series, column: 'column.Column'):
         simple_validation = ~self.validate(series)
         if column.allow_empty:
             # Failing results are those that are not empty, and fail the validation
-            validated = (series.str.len() > 0) & simple_validation
+            validated = ((series.dtype == 'str' and series.str.len() > 0) | (series.dtype == float and not series.isna)) & simple_validation
```
Comment (TMiguelT) [inline pandas_schema/validation.py]: I think this would be clearer and probably more efficient if this were a series of if statements rather than one huge expression

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,5 @@
 git+https://github.com/TMiguelT/sphinxcontrib-restbuilder
 sphinx
 sphinx-autodoc-annotation
+nose
```
Comment (TMiguelT) [inline requirements.txt]: Why is this needed? The tests are written to work with vanilla `unittest`

File: requirements.txt [hunk 3]
```diff
@@ -1,3 +1,5 @@
...
+pandas
```
Comment (TMiguelT) [inline requirements.txt]: This should actually be `.`, which would make it inherit dependencies from the package itself (which is my mistake)

File: test/test_validation.py [hunk 4]
```diff
@@ -583,3 +583,14 @@ def test_custom_message(self):
                 ]
         ), Column('')):
             self.assertRegex(error.message, self.message, 'Validator not using the custom warning message!')
+
+    def test_validating_empty_range_value(self):
```
Comment (TMiguelT) [inline test/test_validation.py]: This test should be in a different class, probably its own class. It certainly has nothing to do with `CustomMessage`.

Also, you should assert something meaningful here; check that it produces the actual results you want, rather than just copying the assertion from above.

**Comment by TMiguelT** [comment]
Hi, thanks for taking a look at this. I can see why this is an annoying bug, but you'll have to fix some things in your PR before I'll accept it

**Comment by caddac** [comment]
@TMiguelT I've updated all requested items. changing the requirements.txt as you requested (and removing/re-installing packages) worked for running tests via `python setup.py test` however to run tests via pycharm I had to manually install nose and pandas into my venv. No big deal but thought I'd mention it.

**Comment by caddac** [comment]
sorry about that, I had 90 set for Apache-Airflow. Fixed it

File: pandas_schema/validation.py [hunk 5]
```diff
@@ -84,7 +84,10 @@ def get_errors(self, series: pd.Series, column: 'column.Column'):
         simple_validation = ~self.validate(series)
         if column.allow_empty:
             # Failing results are those that are not empty, and fail the validation
-            validated = (series.str.len() > 0) & simple_validation
+            if series.dtype == float:
```
Comment (TMiguelT) [inline pandas_schema/validation.py]: I suspect this should be `numpy.issubdtype(series.dtype, numpy.floating)`, but I could be wrong

**Comment by caddac** [comment]
@TMiguelT I'v updated as requested

**Comment by TMiguelT** [comment]
Great! I'll merge this
